---
title: "Project 2 C"
output:
  html_document:
    theme: "darkly"
---

Introduction (40 points)

- 10 points for specific, measurable, and clear scientific question

Scientific Question:
Is the latitudinal correlation between the rates of cutaneous squamous cell carcinoma due to the mutatuions in the p53 gene caused by different levels of UV radiation?

- 10 points for background on the protein/gene/species of interest and where the data is sourced from

Cutaneous squamous cell carcinoma (cSCC) is the second most common form of skin cancer. It is characterized by accelerated growth of squamous cells, which are cells that are found in the tissue that forms the surface of the skin. More than 75% of cSCC and BCC in humans occur on sun-exposed skin. If untreated, cSCC can spread to other parts of the body, leading to serious complications. (DATA FOR THIS SECTION IS FROM THE ___ PAPER)

p53, also called TP53, is a gene that encodes for a nuclear protein involved in controlling cell division and death. It is classified as a tumor suppressor gene, and mutations in p53 have been found in most tumor types. (DATA FOR THIS SECTION IS FROM THE ___ PAPER)

UV ligth is a type of light with shorter wavelengths than visible light. Prolonged exposure to UV radiation from the sun (UVA and UVB radiation), is known to be a risk factor for sunburn, premature aging, eye damage, and all kinds of skin cancers. (DATA FROM THIS SECTION IS FROM THE ___PAPER)

- 10 points for clear, specific, and measurable scientific hypothesis that is in the form of an if-then statement

My hypothesis is that if UV radiation increase the number of P53 mutations leading to squamous cell carcinoma, we would expect to see places with lower latitudes having higher incidence of cSCC, and we should see the UV fingerprint mutations in the P53 genes in cSCC tumor tissue samples.

- 10 points for description of what analyses were done and how the data was downloaded for the project

Loading in Packages (30 points)
```{r}
#if (!requireNamespace("BiocManager", quietly=TRUE))
#install.packages("BiocManager")
#BiocManager::install("msa")
#this package is how I will perform the multiple sequence alignment of the various p53 genes.
library(msa)
#install.packages("ggplot2")
#This package is what I will use to display the UV data by latitude as well as the info for rates of cSCC by latitude
library(ggplot2)
#install.packages("seqinr")
library(seqinr)
#library(biostrings)
library(pdftools)
install.packages("tinytex")
library(tinytex)
```

#- 25 points for definition of each of the packages loaded 

#- 5 points for correctly loading all of the packages needed

Performing Bioinformatics Analysis (50 points)
Step 1: Perform ALignment with the Wild-type p53 gene, as well as the p53 gene with UVA and UVB fingerprint mutations.

```{r}
wt_uva_uvb<-readAAStringSet("wt_uva_uvb.fa.txt" )
class(wt_uva_uvb)
system.file("tex", "texshade.sty", package="msa")
wt_uv_alignment<-msa(wt_uva_uvb, type="protein")
#print(wt_uv_alignment, show="complete")
#msaPrettyPrint(wt_uv_alignment, output="pdf", showNames="none",
#showLogo="none", askForOverwrite=FALSE, verbose=FALSE)
wt_uv_alignment_prettyprint<-pdf_render_page("wt_uv_alignment.pdf", page=1, dpi=300)
png::writePNG(wt_uv_alignment_prettyprint, "wt_uv_alignment_prettyprint.png")
```

![Alignment with WT, UVA and UVB mutated p53](wt_uv_alignment_prettyprint.png)

```{r}
#alignment with data from Clin Canc Res 2015
bioportal_mutations<-readAAStringSet("bioportal_mutations_sequences.txt")
alignment_with_2015_mutations<-msa(bioportal_mutations)
print(alignment_with_2015_mutations, show="complete")
```

```{r}
#alignment with data from MD Anderson Clin Cancer Res 2014
bioportal_mutations_2014<-readAAStringSet("bioportal_mutations_2014.txt")
alignment_with_2014_mutations<-msa(bioportal_mutations_2014)
#print(alignment_with_2014_mutations, show="complete")
#msaPrettyPrint(alignment_with_2014_mutations, output="pdf", showNames="none",
#showLogo="none", askForOverwrite=FALSE, verbose=TRUE,)
#{r, fig.align="center", fig.cap=c("Caption"), echo=FALSE}
#knitr::include_graphics(pretty)
print(msaPrettyPrint(alignment_with_2014_mutations, output="asis", showNames="none",
           showLogo="none", askForOverwrite=FALSE, verbose=FALSE))
```
```{r}
#install.packages("pdftools")
library("pdftools")
image_hope <- pdf_render_page("alignment_with_2014_mutations.pdf", page = 1, dpi = 300)
png::writePNG(image_hope, "page.png")

#![Alignment](page.png)
  
```

![Alignment](page.png)


```{r}
#alignment with data from MD Anderson Clin Cancer Res 2014
bioportal_mutations_2014<-readAAStringSet("bioportal_mutations_2014.txt")
alignment_with_2014_mutations<-msa(bioportal_mutations_2014)
print(alignment_with_2014_mutations, show="complete")
```
Bioinformatics Step #2: Plot the latitude data.

```{r}
library(ggplot2)
library(readxl)
uv_data_morning<-read.csv("uv_data_morning.csv")
uv_data_night<-read.csv("uv_data_night.csv" )
#clean up the data because the CSV files are little messy


clean_night_data<-uv_data_night[c(8, 9, 10, 11),]
clean_night_data<-clean_night_data[, c(1:3)]

clean_morning_data<-uv_data_morning[c(7, 8, 9, 10),]
clean_morning_data<-clean_morning_data[, c(1:3)]
print(clean_morning_data)


colnames(clean_night_data)<-clean_night_data[1,]
clean_night_data<-clean_night_data[-c(1),]

colnames(clean_morning_data)<-clean_morning_data[1,]
clean_morning_data<-clean_morning_data[-c(1),]

print(clean_morning_data)
print(clean_night_data)

#Make the plots using ggplot2
colors <- c("UVA Radiation" = "blue", "UVB Radiation" = "red")
morning_plot<-ggplot(data=clean_morning_data, aes(x=Latitude)) + 
  geom_point(aes(y = UVA, color = "UVA Radiation",))+
  geom_point(aes(y = UVB, color="UVB Radiation", ))+
  labs(color="Legend")+ggtitle("UV Radiation By Latitue-Measured in the Morning") 

night_plot<-ggplot(data=clean_night_data, aes(x=Latitude)) + 
  geom_point(aes(y = UVA, color = "UVA Radiation",))+
  geom_point(aes(y = UVB, color="UVB Radiation", ))+
  labs(color="Legend")+ggtitle("UV Radiation By Latitue-Measured at Night")

print(morning_plot)
print(night_plot)


#check c to have reorder


```
Bioinformatics Step #3 Plot the Skin Cancer Data

```{r}
scc_rates_data<-read.csv("SCC_data.csv")
clean_scc_rates_data<-scc_rates_data[, c(1, 7)]
clean_scc_rates_data<-clean_scc_rates_data[c(3, 5, 7), ]
colnames(clean_scc_rates_data)<-c("Location", "Rate_of_cSCC")

print(clean_scc_rates_data[1,2])

clean_scc_rates_data[1,2]<-gsub("[[:space:]]", "", clean_scc_rates_data[1,2])

clean_scc_rates_data[,2]<-as.numeric(as.character(clean_scc_rates_data[,2]))

#Turn your 'treatment' column into a character vector
clean_scc_rates_data$Location <- as.character(clean_scc_rates_data$Location)
#Then turn it back into a factor with the levels in the correct order
clean_scc_rates_data$Location <- factor(clean_scc_rates_data$Location, levels=unique(clean_scc_rates_data$Location))
print(clean_scc_rates_data)
```


```{r}
library(ggplot2)


cSCC_rate_plot<-ggplot(clean_scc_rates_data, aes(Location, Rate_of_cSCC))+geom_point()

print(cSCC_rate_plot)

```
Bioinformatics Method #4: P-Value
```{r}

#My null hypothesis is that 10% of the samples will have a mutation in the P53 gene.
#From these data, calulcate the sample mean of the percentages.
bioportal_tissue_data<-read.csv("bioportal_tissue_data.csv")
percentages<-bioportal_tissue_data[,4]
percentages<-(percentages)/100
print(percentages)
xbar<-mean(percentages)
print(mean)
a=0.10
#a is our null hypothesis 
std_dev=sd(percentages) 
n=sum(bioportal_tissue_data[,3])
print(n)
p_value_function<-function(xbar_1, a_1, std_dev_1, n_1){
  z_value<-(xbar_1-a_1)/(std_dev_1/sqrt(n_1))
  p_value<-2*pnorm(-abs(z_value))
  return(p_value)
}
p_value_function(xbar, a, std_dev, n)
  #z=(xbar-a)/(std_dev/sqrt(n))
#print(z)
#p_value=2*pnorm(-abs(z))
#print(p_value)
```
Based on this small value, we can reject the null hypothesis and say that someone cSCC does not have a P53 mutation. 